### ISSUE TITLE
Normalize types in perf suite

### ISSUE THREAD

**Issue by jdinan** [Sandia-OpenSHMEM/SOS#669]
Make `start_len` and `max_len` in both `latency_common` and `bw_common` unsigned long int data type to make it unified.  Check if any other types in the perf metrics structure should also be normalized.

**Comment by jdinan** [comment]
Noticed some additional warnings:
```
In file included from ../../../../test/performance/shmem_perf_suite/shmem_bw_put_ctx_perf.c:40:0:
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h: In function ‘uni_bw_ctx’:
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:63:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->warmup; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:64:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (j = 0; j < metric_info->window_size; j++) {
                               ^
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:93:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->trials; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:94:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (j = 0; j < metric_info->window_size; j++) {
                               ^
  CCLD     shmem_bw_put_ctx_perf
  CC       shmem_bw_put_ctx_perf_nb-shmem_bw_put_ctx_perf.o
In file included from ../../../../test/performance/shmem_perf_suite/shmem_bw_put_ctx_perf.c:40:0:
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h: In function ‘uni_bw_ctx’:
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:63:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->warmup; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:64:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (j = 0; j < metric_info->window_size; j++) {
                               ^
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:93:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->trials; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/uni_dir_ctx.h:94:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (j = 0; j < metric_info->window_size; j++) {
                               ^
  CCLD     shmem_bw_put_ctx_perf_nb
  CC       shmem_bibw_put_ctx_perf-shmem_bibw_put_ctx_perf.o
In file included from ../../../../test/performance/shmem_perf_suite/shmem_bibw_put_ctx_perf.c:42:0:
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h: In function ‘bi_bw_ctx’:
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:61:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (i = 0; i < metric_info->warmup; i++) {
                       ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:62:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for(j = 0; j < metric_info->window_size; j++) {
                          ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:89:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->trials; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:90:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for(j = 0; j < metric_info->window_size; j++) {
                              ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:110:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->trials; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:111:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for(j = 0; j < metric_info->window_size; j++) {
                              ^
  CCLD     shmem_bibw_put_ctx_perf
  CC       shmem_bibw_put_ctx_perf_nb-shmem_bibw_put_ctx_perf.o
In file included from ../../../../test/performance/shmem_perf_suite/shmem_bibw_put_ctx_perf.c:42:0:
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h: In function ‘bi_bw_ctx’:
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:61:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (i = 0; i < metric_info->warmup; i++) {
                       ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:62:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for(j = 0; j < metric_info->window_size; j++) {
                          ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:89:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->trials; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:90:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for(j = 0; j < metric_info->window_size; j++) {
                              ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:110:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (i = 0; i < metric_info->trials; i++) {
                           ^
../../../../test/performance/shmem_perf_suite/bi_dir_ctx.h:111:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for(j = 0; j < metric_info->window_size; j++) {
                              ^
  CCLD     shmem_bibw_put_ctx_perf_nb
```

**Comment by wrrobin** [comment]
Thanks. Will fix these.

**Comment by jdinan** [comment]
Thanks -- Figured it's better to leave these alone, since you might fix them anyway in the course of updating the types.

**Comment by wrrobin** [comment]
Yes, I also thought so earlier. But, I have just created the pull request for this. We can merge all together may be.

**Comment by wrrobin** [comment]
Closed by PR #693
